### PR TITLE
Analyzing information_schema based on window: startTime and endTime

### PR DESCRIPTION
**Pull request by sridivakar**
with these command line options:
```bash
--read_from_info_schema 
--read_from_info_schema_start_time "2023-08-15 09:00:00" 
--read_from_info_schema_end_time "2023-08-16"
--read_from_info_schema_timeout_in_secs 60  
--info_schema_table_name `region-us`.INFORMATION_SCHEMA.JOBS 
--processing_project_id <my-project>
--output_table <my-project>.<my-dataset>.antipattern_output_table
```

Query on INFORMATION SCHEMA:
```sql
SELECT
  project_id,
  CONCAT(project_id, ":US.",  job_id) job_id, 
  query, 
  total_slot_ms / (1000 * 60 * 60 ) AS slot_hours, 
  user_email 
FROM
`region-us`.INFORMATION_SCHEMA.JOBS
WHERE 
  creation_time BETWEEN '2023-08-15 09:00:00' AND '2023-08-16'
  AND total_slot_ms > 0
  AND (statement_type != "SCRIPT" OR statement_type IS NULL)
  AND (reservation_id != 'default-pipeline' or reservation_id IS NULL)
  AND query not like '%INFORMATION_SCHEMA%' 
ORDER BY 
  project_id, start_time desc
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -225,6 +225,22 @@ Must be set along with `--read_from_info_schema`. <br>
...
+to specify start and end date or timestamp of INFORMATION_SCHEMA to read.<br>
+Must be set along with `--read_from_info_schema`. <br>
+Defaults to `--read_from_info_schema_days` option.
+</ul>
+
+`--read_from_info_schema_timeout_in_secs=n`
+<ul>
+Specifies timeout, in secs, to query INFORMATION SCHEMA<br>
+Must be set along with `--read_from_info_schema`. <br>
+Defaults to 10.
```
Comment (franklinWhaite) [inline README.md:241]: Could we default it to something higher, maybe 60?

Comment (sridivakar) [reply README.md:241]: The rationale for 10 seconds is based on the default timeout for the QueryJob (Ref: 
https://cloud.google.com/bigquery/docs/reference/rest/v2/jobs/query#:~:text=By%20default%2C%20this%20limit%20is,period%20in%20the%20timeoutMs%20field.
).

sure, will change it to 60 seconds.

File: bigquery-antipattern-recognition/src/main/java/com/google/zetasql/toolkit/antipattern/cmd/InformationSchemaQueryIterable.java [hunk 2]
```diff
@@ -19,26 +19,33 @@
...
 import java.util.Iterator;
+import java.util.Optional;
 
 public class InformationSchemaQueryIterable implements Iterator<InputQuery> {
 
   Iterator<FieldValueList> fieldValueListIterator;
   String IS_TABLE_DEFAULT = "`region-us`.INFORMATION_SCHEMA.JOBS";
   String DAYS_BACK_DEFAULT = "30";
   Integer SLOTMS_MIN_DEFAULT = 0;
+  Long TIMEOUT_SECS_DEFAULT = 10L;
```
[the remark this answers is not in the document]
Comment (sridivakar) [reply bigquery-antipattern-recognition/src/main/java/com/google/zetasql/toolkit/antipattern/cmd/InformationSchemaQueryIterable.java]: Addressed, increased to 60 seconds

**Comment by sridivakar** [comment]
> LGTM, I only left one comment on the timeout, once that is addressed we can merge

Thanks, addressed the comment.

**Review by franklinWhaite — APPROVED** [review]

**Pull request merged**